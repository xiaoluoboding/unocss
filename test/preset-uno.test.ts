import { createGenerator, escapeSelector } from '@unocss/core'
import presetUno from '@unocss/preset-uno'
import { expect, test } from 'vitest'

const targets = [
  '-gap-y-5',
  '-m-auto',
  '!hover:px-10',
  '!p-5px',
  'all:m-auto',
  'bg-auto',
  'bg-blend-normal',
  'bg-blend-color-burn',
  'bg-blend-luminosity',
  'bg-bottom',
  'bg-clip-border',
  'bg-clip-text',
  'bg-cover',
  'bg-fixed',
  'bg-gradient-to-t',
  'bg-gradient-to-tl',
  'bg-hex-452233/40',
  'bg-local',
  'bg-no-repeat',
  'bg-none',
  'bg-origin-border',
  'bg-opacity-45',
  'bg-repeat-space',
  'bg-right-bottom',
  'bg-scroll',
  // colors
  'color-blue',
  'color-blue-400',
  'color-blue-400/10',
  'color-blue/10',
  'color-blue-gray',
  'color-blue-gray-400',
  'color-bluegray',
  'color-bluegray-400',
  'color-blue-gray-400/10',
  'color-blue-gray/10',
  'color-bluegray-400/10',
  'color-bluegray/10',
  // fill
  'fill-none',
  'fill-current',
  'fill-green-400',
  'fill-opacity-80',
  'fill-[#123]',
  'fill-[1rem]',
  // stroke
  'stroke-none',
  'stroke-current',
  'stroke-green-400',
  'stroke-opacity-80',
  'stroke-[#123]',
  'stroke-[1rem]',
  'stroke-size-1',
  'stroke-size-1px',
  'stroke-size-[1rem]',
  'caption-top',
  'caption-bottom',
  'children:m-auto',
  'dark:not-odd:text-red',
  'dark:text-xl',
  '.dark:text-xl',
  '@dark:text-xl',
  'duration-111',
  'flex-[hi]',
  'flex',
  'font-mono',
  'from-current',
  'from-green-500',
  'from-transparent',
  'gap-4',
  'gap-x-1',
  'hover:!p-1',
  'inline-table',
  'first:p-2',
  'not-hover:p-3',
  'group-focus:p-4',
  'group-hover:group-focus:text-center',
  'peer-checked:bg-blue-500',
  'peer-not-placeholder-shown:text-2xl',
  'hover:not-first:checked:bg-red/10',
  'hover:p-5',
  'leading-2',
  'light:text-sm',
  'm-[3em]',
  'm-0',
  'm-1/2',
  'm-auto',
  '-mb-px',
  'aspect-ratio-auto',
  'aspect-ratio-3/2',
  'aspect-ratio-0.7',
  'aspect-ratio-$var',
  'aspect-ratio-[auto_16/9]',
  'md:!hidden',
  'md:m-1',
  'mix-blend-normal',
  'mix-blend-color-light',
  'my-auto',
  'op-10',
  'opacity-0',
  'opacity-$opa',
  'order-first',
  'overflow-auto',
  'overflow-x-scroll',
  'overscroll-x-auto',
  'overscroll-contain',
  'p-2',
  'p-t-2',
  'p2',
  'pl-10px',
  'pt-2',
  'pt2',
  '-p-px',
  'sm:m-1',
  'sm:m1',
  'lt-sm:m1',
  'lt-lg:m2',
  'at-sm:m1',
  'at-lg:m2',
  'at-2xl:m2',
  'table',
  'table-auto',
  'table-caption',
  'table-empty-cells-visible',
  'table-empty-cells-hidden',
  'table-footer-group',
  'table-row-group',
  'text-[#124]',
  'text-[2em]',
  'text-[calc(1em-1px)]',
  'text-4xl',
  'text-base',
  'text-black/10',
  'text-rose',
  'text-lg',
  'text-red-100',
  'text-red-200/10',
  'text-red-300/20',
  'text-red100',
  'text-red2',
  'text-[var(--color)]',
  'to-current',
  'to-green-500',
  'to-transparent',
  'top-0',
  'top-$top-height',
  'tracking-wide',
  'word-spacing-wide',
  'word-spacing-2',
  '-z-1',
  'z-0',
  'z-1',
  'z-100',
  'box-content',
  'box-border',
  'shadow',
  'shadow-transparent',
  'shadow-current',
  'shadow-none',
  'shadow-xl',
  'shadow-green-500',
  'mt-[-10.2%]',
  'ring',
  'ring-10',
  'ring-red2',
  'ring-offset-4',
  'ring-offset-green5',
  'inset-x-5',
  'text-opacity-[13.3333333%]',
  'via-current',
  'via-green-500',
  'via-green-500/30',
  'via-transparent',
  'via-opacity-30',
  'list-none',
  'list-disc',
  'list-outside',
  'list-none-inside',
  'image-render-pixel',
  'decoration-slice',
  'appearance-none',
  'accent-op-90',
  'accent-red',
  'caret-op-90',
  'caret-red',
  'space-x-2',
  'space-y-4',
  '-space-x-4',
  'space-x-reverse',
  'next:mt-0',
  'italic',
  'lining-nums',
  'normal-nums',
  'tabular-nums',
  'antialiased',
  'hyphens-none',
  'tab',
  'tab-6',
  'tab-inherit',
  'indent',
  'indent-1/2',
  'indent-lg',
  'text-stroke-6',
  'text-stroke-sm',
  'text-stroke-blue-500',
  'text-stroke-op-60',
  'text-shadow',
  'text-shadow-lg',
  'text-shadow-none',
  'text-shadow-$var',
  'write-normal',
  'write-orient-sideways',
  'color-$red',
  'tab-$tabprop',
  'items-$size',
  'object-$fit',
  'bg-$test-variable',
  'bg-blend-$data',
  '!m-$c-m',
  'mt-$height',
  'pt-$title2',
  'space-x-$space',
  'outline-solid',
  'outline-color-red-1',
  'outline-width-10px',
  'outline-inset',
  'outline-110',
  'outline-blue-2',
  'outline-none',
  'outline-[var(--red)]',
  'outline-size-[var(--width)]',
  'outline-offset-[var(--offset)]',
  'placeholder-color-red-1',
  'placeholder-transparent',
  'placeholder-opacity-60',
  'select-none',
  'ws-nowrap',
  'b-2',
  'v-top',
  'v-mid',
  'disabled:op50',
  'touch-pan-left',
  'touch-pan-y',
  'columns-3',
  'columns-5em',
  'columns-auto',
  'will-change-transform',

  // scroll-spaces
  'scroll-m-[3em]',
  'scroll-m-0',
  'scroll-m-1/2',
  'scroll-m-auto',
  '-scroll-mb-px',
  'scroll-p-2',
  'scroll-p-t-2',
  'scroll-p2',
  'scroll-pl-10px',
  'scroll-pt-2',
  'scroll-pt2',
  '-scroll-p-px',

  // custom colors
  'text-custom-a',
  'bg-custom-b',
  'border-custom-b/10',
]

const nonTargets = [
  '--p-2',
  'before:before:m2',
  'hi',
  'row-{row.id}',
  'tabs',
  'tab.hello',
  'text-anything',
  'p-anything',
  'rotate-[3]deg',
  'will-change-all',
  'will-change-none',
]

const uno = createGenerator({
  presets: [
    presetUno({
      dark: 'media',
    }),
  ],
  theme: {
    colors: {
      custom: {
        a: 'var(--custom)',
        b: 'rgba(var(--custom), %alpha)',
      },
    },
  },
})

test('targets', async() => {
  const code = targets.join(' ')
  const { css } = await uno.generate(code)
  const { css: css2 } = await uno.generate(code)

  const unmatched = []
  for (const i of targets) {
    if (!css.includes(escapeSelector(i)))
      unmatched.push(i)
  }
  expect(unmatched).toEqual([])
  expect(css).toMatchSnapshot()
  expect(css).toEqual(css2)
})

test('non-targets', async() => {
  const code = nonTargets.join(' ')
  const { css, matched } = await uno.generate(code)

  expect(Array.from(matched)).toEqual([])
  expect(css).toBe('')
})
