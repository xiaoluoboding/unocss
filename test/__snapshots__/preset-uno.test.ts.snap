// Vitest Snapshot v1

exports[`targets 1`] = `
"/* layer: default */
.bg-\\\\$test-variable{background-color:var(--test-variable);}
.color-\\\\$red{color:var(--red);}
.items-\\\\$size{align-items:var(--size);}
.object-\\\\$fit{object-fit:var(--fit);}
.tab-\\\\$tabprop{tab-size:var(--tabprop);}
.bg-blend-\\\\$data{background-blend-mode:var(--data);}
.top-\\\\$top-height{top:var(--top-height);}
.top-0{top:0rem;}
.inset-x-5{left:1.25rem;right:1.25rem;}
.-z-1{z-index:-1;}
.z-0{z-index:0;}
.z-1{z-index:1;}
.z-100{z-index:100;}
.order-first{order:-9999;}
.-m-auto,.all\\\\:m-auto *,.children\\\\:m-auto>*,.m-auto{margin:auto;}
.\\\\!m-\\\\$c-m{margin:var(--c-m) !important;}
.m-\\\\[3em\\\\]{margin:3em;}
.m-0{margin:0rem;}
.m-1\\\\/2{margin:50%;}
.my-auto{margin-top:auto;margin-bottom:auto;}
.-mb-px{margin-bottom:-1px;}
.mt-\\\\[-10\\\\.2\\\\%\\\\]{margin-top:-10.2%;}
.mt-\\\\$height{margin-top:var(--height);}
.next\\\\:mt-0+*{margin-top:0rem;}
.box-border{box-sizing:border-box;}
.box-content{box-sizing:content-box;}
.aspect-ratio-auto{aspect-ratio:auto;}
.aspect-ratio-\\\\[auto_16\\\\/9\\\\]{aspect-ratio:auto 16/9;}
.aspect-ratio-\\\\$var{aspect-ratio:var(--var);}
.aspect-ratio-0\\\\.7{aspect-ratio:0.7;}
.aspect-ratio-3\\\\/2{aspect-ratio:3/2;}
.flex{display:flex;}
.flex-\\\\[hi\\\\]{flex:hi;}
.caption-top{caption-side:top;}
.caption-bottom{caption-side:bottom;}
.inline-table{display:inline-table;}
.table{display:table;}
.table-auto{table-layout:auto;}
.table-empty-cells-visible{empty-cells:show;}
.table-empty-cells-hidden{empty-cells:hide;}
.table-caption{display:table-caption;}
.table-footer-group{display:table-footer-group;}
.table-row-group{display:table-row-group;}
.touch-pan-left,.touch-pan-y{--un-pan-x:var(--un-empty,/*!*/ /*!*/);--un-pan-y:var(--un-empty,/*!*/ /*!*/);--un-pinch-zoom:var(--un-empty,/*!*/ /*!*/);--un-touch-action:var(--un-pan-x) var(--un-pan-y) var(--un-pinch-zoom);}
.touch-pan-left{--un-pan-x:pan-left;touch-action:var(--un-touch-action);}
.touch-pan-y{--un-pan-y:pan-y;touch-action:var(--un-touch-action);}
.select-none{user-select:none;}
.scroll-m-\\\\[3em\\\\]{scroll-margin:3em;}
.scroll-m-0{scroll-margin:0rem;}
.scroll-m-1\\\\/2{scroll-margin:50%;}
.scroll-m-auto{scroll-margin:auto;}
.-scroll-mb-px{scroll-margin-bottom:-1px;}
.-scroll-p-px{scroll-padding:-1px;}
.scroll-p-2,.scroll-p2{scroll-padding:0.5rem;}
.scroll-p-t-2,.scroll-pt-2,.scroll-pt2{scroll-padding-top:0.5rem;}
.scroll-pl-10px{scroll-padding-left:10px;}
.list-disc{list-style-type:disc;}
.list-none{list-style-type:none;}
.list-none-inside{list-style-position:inside;list-style-type:none;}
.list-outside{list-style-position:outside;}
.appearance-none{appearance:none;-webkit-appearance:none;}
.columns-3{column:3;}
.columns-5em{column:5em;}
.columns-auto{column:auto;}
.gap-4{grid-gap:1rem;gap:1rem;}
.gap-x-1{grid-column-gap:0.25rem;column-gap:0.25rem;}
.-gap-y-5{grid-row-gap:-1.25rem;row-gap:-1.25rem;}
.-space-x-4>:not([hidden])~:not([hidden]){--un-space-x-reverse:0;margin-left:calc(-1rem * calc(1 - var(--un-space-x-reverse)));margin-right:calc(-1rem * var(--un-space-x-reverse));}
.space-x-\\\\$space>:not([hidden])~:not([hidden]){--un-space-x-reverse:0;margin-left:calc(var(--space) * calc(1 - var(--un-space-x-reverse)));margin-right:calc(var(--space) * var(--un-space-x-reverse));}
.space-x-2>:not([hidden])~:not([hidden]){--un-space-x-reverse:0;margin-left:calc(0.5rem * calc(1 - var(--un-space-x-reverse)));margin-right:calc(0.5rem * var(--un-space-x-reverse));}
.space-x-reverse>:not([hidden])~:not([hidden]){--un-space-x-reverse:1;}
.space-y-4>:not([hidden])~:not([hidden]){--un-space-y-reverse:0;margin-top:calc(1rem * calc(1 - var(--un-space-y-reverse)));margin-bottom:calc(1rem * var(--un-space-y-reverse));}
.overflow-auto{overflow:auto;}
.overflow-x-scroll{overflow-x:scroll;}
.overscroll-contain{overscroll-behavior:contain;}
.overscroll-x-auto{overscroll-behavior-x:auto;}
.ws-nowrap{white-space:nowrap;}
.b-2{border-width:2px;border-style:solid;}
.border-custom-b\\\\/10{border-color:rgba(var(--custom), 0.1);}
.bg-custom-b{background-color:rgba(var(--custom), 1);}
.bg-hex-452233\\\\/40{background-color:rgba(69,34,51,0.4);}
.hover\\\\:not-first\\\\:checked\\\\:bg-red\\\\/10:hover:not(:first-child):checked{background-color:rgba(248,113,113,0.1);}
.peer:checked~.peer-checked\\\\:bg-blue-500{--un-bg-opacity:1;background-color:rgba(59,130,246,var(--un-bg-opacity));}
.bg-opacity-45{--un-bg-opacity:0.45;}
.decoration-slice{box-decoration-break:slice;}
.bg-fixed{background-attachment:fixed;}
.bg-local{background-attachment:local;}
.bg-scroll{background-attachment:scroll;}
.bg-blend-normal{background-blend-mode:normal;}
.bg-blend-color-burn{background-blend-mode:color-burn;}
.bg-blend-luminosity{background-blend-mode:luminosity;}
.bg-clip-border{-webkit-background-clip:border-box;background-attachment:border-box;}
.bg-clip-text{-webkit-background-clip:text;background-attachment:text;}
.from-current{--un-gradient-from:currentColor;--un-gradient-stops:var(--un-gradient-from), var(--un-gradient-to, rgba(255, 255, 255, 0));}
.from-green-500{--un-gradient-from:rgba(34,197,94, var(--un-from-opacity, 1));--un-gradient-stops:var(--un-gradient-from), var(--un-gradient-to, rgba(255, 255, 255, 0));}
.from-transparent{--un-gradient-from:transparent;--un-gradient-stops:var(--un-gradient-from), var(--un-gradient-to, rgba(255, 255, 255, 0));}
.to-current{--un-gradient-to:currentColor;}
.to-green-500{--un-gradient-to:rgba(34,197,94, var(--un-to-opacity, 1));}
.to-transparent{--un-gradient-to:transparent;}
.via-current{--un-gradient-stops:var(--un-gradient-from), currentColor, var(--un-gradient-to, rgba(255, 255, 255, 0));}
.via-green-500{--un-gradient-stops:var(--un-gradient-from), rgba(34,197,94, var(--un-via-opacity, 1)), var(--un-gradient-to, rgba(255, 255, 255, 0));}
.via-green-500\\\\/30{--un-gradient-stops:var(--un-gradient-from), rgba(34,197,94,0.3), var(--un-gradient-to, rgba(255, 255, 255, 0));}
.via-transparent{--un-gradient-stops:var(--un-gradient-from), transparent, var(--un-gradient-to, rgba(255, 255, 255, 0));}
.via-opacity-30{--un-via-opacity:0.3;}
.bg-none{background-image:none;}
.bg-gradient-to-t{background-image:linear-gradient(to top, var(--un-gradient-stops));}
.bg-gradient-to-tl{background-image:linear-gradient(to top left, var(--un-gradient-stops));}
.bg-origin-border{background-origin:border-box;}
.bg-bottom{background-position:bottom;}
.bg-right-bottom{background-position:right bottom;}
.bg-no-repeat{background-repeat:no-repeat;}
.bg-repeat-space{background-position:space;}
.bg-auto{background-size:auto;}
.bg-cover{background-size:cover;}
.fill-\\\\[\\\\#123\\\\]{--un-fill-opacity:1;fill:rgba(17,34,51,var(--un-fill-opacity));}
.fill-\\\\[1rem\\\\]{fill:1rem;}
.fill-current{fill:currentColor;}
.fill-green-400{--un-fill-opacity:1;fill:rgba(74,222,128,var(--un-fill-opacity));}
.fill-opacity-80{--un-fill-opacity:0.8;}
.fill-none{fill:none;}
.stroke-size-\\\\[1rem\\\\]{stroke-width:1rem;}
.stroke-size-1,.stroke-size-1px{stroke-width:1px;}
.stroke-\\\\[\\\\#123\\\\]{--un-stroke-opacity:1;stroke:rgba(17,34,51,var(--un-stroke-opacity));}
.stroke-\\\\[1rem\\\\]{stroke:1rem;}
.stroke-current{stroke:currentColor;}
.stroke-green-400{--un-stroke-opacity:1;stroke:rgba(74,222,128,var(--un-stroke-opacity));}
.stroke-opacity-80{--un-stroke-opacity:0.8;}
.stroke-none{stroke:none;}
.-p-px{padding:-1px;}
.\\\\!p-5px{padding:5px !important;}
.first\\\\:p-2:first-child,.p-2,.p2{padding:0.5rem;}
.group:focus .group-focus\\\\:p-4{padding:1rem;}
.hover\\\\:\\\\!p-1:hover{padding:0.25rem !important;}
.hover\\\\:p-5:hover{padding:1.25rem;}
.not-hover\\\\:p-3:not(:hover){padding:0.75rem;}
.\\\\!hover\\\\:px-10:hover{padding-left:2.5rem !important;padding-right:2.5rem !important;}
.p-t-2,.pt-2,.pt2{padding-top:0.5rem;}
.pl-10px{padding-left:10px;}
.pt-\\\\$title2{padding-top:var(--title2);}
.group:focus:hover .group-hover\\\\:group-focus\\\\:text-center{text-align:center;}
.indent{text-indent:1.5rem;}
.indent-1\\\\/2{text-indent:50%;}
.indent-lg{text-indent:2rem;}
.v-mid{vertical-align:middle;}
.v-top{vertical-align:top;}
.font-mono{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,\\"Liberation Mono\\",\\"Courier New\\",monospace;}
.dark .\\\\.dark\\\\:text-xl{font-size:1.25rem;line-height:1.75rem;}
.peer:not(:placeholder-shown)~.peer-not-placeholder-shown\\\\:text-2xl{font-size:1.5rem;line-height:2rem;}
.text-4xl{font-size:2.25rem;line-height:2.5rem;}
.text-base{font-size:1rem;line-height:1.5rem;}
.text-lg{font-size:1.125rem;line-height:1.75rem;}
.leading-2{line-height:0.5rem;}
.tracking-wide{letter-spacing:0.025em;}
.word-spacing-2{word-spacing:0.5rem;}
.word-spacing-wide{word-spacing:0.025em;}
.italic{font-style:italic;}
.lining-nums,.tabular-nums{--un-ordinal:var(--un-empty,/*!*/ /*!*/);--un-slashed-zero:var(--un-empty,/*!*/ /*!*/);--un-numeric-figure:var(--un-empty,/*!*/ /*!*/);--un-numeric-spacing:var(--un-empty,/*!*/ /*!*/);--un-numeric-fraction:var(--un-empty,/*!*/ /*!*/);font-variant-numeric:var(--un-ordinal) var(--un-slashed-zero) var(--un-numeric-figure) var(--un-numeric-spacing) var(--un-numeric-fraction);}
.lining-nums{--un-numeric-figure:lining-nums;}
.tabular-nums{--un-numeric-spacing:tabular-nums;}
.normal-nums{font-variant-numeric:normal;}
.color-blue,.color-blue-400{--un-text-opacity:1;color:rgba(96,165,250,var(--un-text-opacity));}
.color-blue-400\\\\/10,.color-blue\\\\/10{color:rgba(96,165,250,0.1);}
.color-blue-gray,.color-blue-gray-400,.color-bluegray,.color-bluegray-400{--un-text-opacity:1;color:rgba(148,163,184,var(--un-text-opacity));}
.color-blue-gray-400\\\\/10,.color-blue-gray\\\\/10,.color-bluegray-400\\\\/10,.color-bluegray\\\\/10{color:rgba(148,163,184,0.1);}
.text-\\\\[\\\\#124\\\\]{--un-text-opacity:1;color:rgba(17,34,68,var(--un-text-opacity));}
.text-\\\\[2em\\\\]{color:2em;}
.text-\\\\[calc\\\\(1em-1px\\\\)\\\\]{color:calc(1em - 1px);}
.text-\\\\[var\\\\(--color\\\\)\\\\]{color:var(--color);}
.text-black\\\\/10{color:rgba(0,0,0,0.1);}
.text-custom-a{color:var(--custom);}
.text-red-100,.text-red100{--un-text-opacity:1;color:rgba(254,226,226,var(--un-text-opacity));}
.text-red-200\\\\/10{color:rgba(254,202,202,0.1);}
.text-red-300\\\\/20{color:rgba(252,165,165,0.2);}
.text-red2{--un-text-opacity:1;color:rgba(254,202,202,var(--un-text-opacity));}
.text-rose{--un-text-opacity:1;color:rgba(251,113,133,var(--un-text-opacity));}
.text-opacity-\\\\[13\\\\.3333333\\\\%\\\\]{--un-text-opacity:13.3333333%;}
.antialiased{-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;font-smoothing:grayscale;}
.tab{-moz-tab-size:4;-o-tab-size:4;tab-size:4;}
.tab-6{-moz-tab-size:6;-o-tab-size:6;tab-size:6;}
.tab-inherit{-moz-tab-size:inherit;-o-tab-size:inherit;tab-size:inherit;}
.text-stroke-6{-webkit-text-stroke-width:6px;}
.text-stroke-sm{-webkit-text-stroke-width:thin;}
.text-stroke-blue-500{--un-text-stroke-opacity:1;-webkit-text-stroke-color:rgba(59,130,246,var(--un-text-stroke-opacity));}
.text-stroke-op-60{--un-text-stroke-opacity:0.6;}
.text-shadow{text-shadow:0px 0px 1px rgb(0 0 0/20%), 0px 0px 1px rgb(1 0 5/10%);}
.text-shadow-\\\\$var{text-shadow:var(--var);}
.text-shadow-lg{text-shadow:3px 3px 6px rgb(0 0 0 / 26%), 0 0 5px rgb(15 3 86 / 22%);}
.text-shadow-none{text-shadow:none;}
.hyphens-none{-webkit-hyphens:none;-ms-hyphens:none;hyphens:none;}
.write-normal{writing-mode:horizontal-tb;}
.write-orient-sideways{text-orientation:sideways;}
.placeholder-opacity-60{placeholder-opacity:0.6;}
.placeholder-color-red-1{--un-placeholder-color-opacity:1;placeholder-color:rgba(254,226,226,var(--un-placeholder-color-opacity));}
.placeholder-transparent{placeholder-color:transparent;}
.caret-red{--un-caret-opacity:1;caret-color:rgba(248,113,113,var(--un-caret-opacity));}
.caret-op-90{--un-caret-opacity:0.9;}
.accent-red{--un-accent-opacity:1;accent-color:rgba(248,113,113,var(--un-accent-opacity));}
.accent-op-90{--un-accent-opacity:0.9;}
.disabled\\\\:op50:disabled{opacity:0.5;}
.op-10{opacity:0.1;}
.opacity-\\\\$opa{opacity:var(--opa);}
.opacity-0{opacity:0;}
.mix-blend-normal{mix-blend-mode:normal;}
.mix-blend-color-light{mix-blend-mode:color-light;}
.shadow{--un-shadow-color:0,0,0;--un-shadow:0 1px 3px 0 rgba(var(--un-shadow-color), 0.1), 0 1px 2px 0 rgba(var(--un-shadow-color), 0.06);box-shadow:var(--un-ring-offset-shadow, 0 0 #0000), var(--un-ring-shadow, 0 0 #0000), var(--un-shadow);}
.shadow-current{--un-shadow-color:currentColor;}
.shadow-green-500{--un-shadow-color:34,197,94;}
.shadow-none{--un-shadow-color:0,0,0;--un-shadow:none;box-shadow:var(--un-ring-offset-shadow, 0 0 #0000), var(--un-ring-shadow, 0 0 #0000), var(--un-shadow);}
.shadow-transparent{--un-shadow-color:transparent;}
.shadow-xl{--un-shadow-color:0,0,0;--un-shadow:0 20px 25px -5px rgba(var(--un-shadow-color), 0.1), 0 10px 10px -5px rgba(var(--un-shadow-color), 0.04);box-shadow:var(--un-ring-offset-shadow, 0 0 #0000), var(--un-ring-shadow, 0 0 #0000), var(--un-shadow);}
.outline-none{outline:2px solid transparent;outline-offset:2px;}
.outline-\\\\[var\\\\(--red\\\\)\\\\]{outline-color:var(--red);}
.outline-110{outline-width:27.5rem;}
.outline-blue-2{--un-outline-color-opacity:1;outline-color:rgba(191,219,254,var(--un-outline-color-opacity));}
.outline-color-red-1{--un-outline-color-opacity:1;outline-color:rgba(254,226,226,var(--un-outline-color-opacity));}
.outline-inset{outline-style:inset;}
.outline-offset-\\\\[var\\\\(--offset\\\\)\\\\]{outline-offset:var(--offset);}
.outline-size-\\\\[var\\\\(--width\\\\)\\\\]{outline-width:var(--width);}
.outline-solid{outline-style:solid;}
.outline-width-10px{outline-width:10px;}
.ring{--un-ring-inset:var(--un-empty, );--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-color:rgba(59, 130, 246, .5);--un-ring-offset-shadow:var(--un-ring-inset) 0 0 0 var(--un-ring-offset-width) var(--un-ring-offset-color);--un-ring-shadow:var(--un-ring-inset) 0 0 0 calc(1px + var(--un-ring-offset-width)) var(--un-ring-color);-webkit-box-shadow:var(--un-ring-offset-shadow), var(--un-ring-shadow), var(--un-shadow, 0 0 #0000);box-shadow:var(--un-ring-offset-shadow), var(--un-ring-shadow), var(--un-shadow, 0 0 #0000);}
.ring-10{--un-ring-inset:var(--un-empty, );--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-color:rgba(59, 130, 246, .5);--un-ring-offset-shadow:var(--un-ring-inset) 0 0 0 var(--un-ring-offset-width) var(--un-ring-offset-color);--un-ring-shadow:var(--un-ring-inset) 0 0 0 calc(10px + var(--un-ring-offset-width)) var(--un-ring-color);-webkit-box-shadow:var(--un-ring-offset-shadow), var(--un-ring-shadow), var(--un-shadow, 0 0 #0000);box-shadow:var(--un-ring-offset-shadow), var(--un-ring-shadow), var(--un-shadow, 0 0 #0000);}
.ring-offset-4{--un-ring-offset-width:4px;}
.ring-red2{--un-ring-opacity:1;--un-ring-color:rgba(254,202,202,var(--un-ring-opacity));}
.ring-offset-green5{--un-ring-offset-opacity:1;--un-ring-offset-color:rgba(34,197,94,var(--un-ring-offset-opacity));}
.image-render-pixel{-ms-interpolation-mode:nearest-neighbor;image-rendering:-webkit-optimize-contrast;image-rendering:-moz-crisp-edges;image-rendering:-o-pixelated;image-rendering:pixelated;}
.duration-111{transition-duration:111ms;}
.will-change-transform{will-change:transform;}
@media (prefers-color-scheme: dark){
.\\\\@dark\\\\:text-xl,.dark\\\\:text-xl{font-size:1.25rem;line-height:1.75rem;}
.dark\\\\:not-odd\\\\:text-red:not(:nth-child(odd)){--un-text-opacity:1;color:rgba(248,113,113,var(--un-text-opacity));}
}
@media (prefers-color-scheme: light){
.light\\\\:text-sm{font-size:0.875rem;line-height:1.25rem;}
}
@media (max-width: 1024px){
.lt-lg\\\\:m2{margin:0.5rem;}
}
@media (max-width: 640px){
.lt-sm\\\\:m1{margin:0.25rem;}
}
@media (min-width: 640px){
.sm\\\\:m-1,.sm\\\\:m1{margin:0.25rem;}
}
@media (min-width: 640px) and (max-width: 768px){
.at-sm\\\\:m1{margin:0.25rem;}
}
@media (min-width: 768px){
.md\\\\:m-1{margin:0.25rem;}
.md\\\\:\\\\!hidden{display:none !important;}
}
@media (min-width: 1024px) and (max-width: 1280px){
.at-lg\\\\:m2{margin:0.5rem;}
}
@media (min-width: 1536px){
.at-2xl\\\\:m2{margin:0.5rem;}
}"
`;
